src/v/iceberg/values_json.h

```diff
@@ -0,0 +1,21 @@
+// Copyright 2025 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+#pragma once
+
+#include "iceberg/datatypes.h"
+#include "iceberg/json_writer.h"
+#include "iceberg/values.h"
+#include "json/document.h"
+
+namespace iceberg {
+std::optional<value>
+value_from_json(const json::Value&, const field_type&, field_required);
```

Comment:
Should I be worried about using `json::Value`? There are oversized alloc concerns here for both container types and strings.

Reply:
yeah i guess that's more salient for values than schemas or types or whatever. is there an alternative? or you're suggesting that we might need to build something more complicated?

Reply:
I just don't want to crash because of an overside alloc. Where will this be used? That might help make me feel better until we create a custom JSON parser/serializer.